src/build.py

```diff
@@ -1529,10 +1530,7 @@ def ExecuteEmscriptenTestSuite(name, config, outdir, warn_only):
          'binaryen2', '--em-config', config],
         cwd=outdir)
   except proc.CalledProcessError:
-    if warn_only:
-      buildbot.Warn()
-    else:
-      buildbot.Fail()
+    buildbot.FailUnless(lambda: warn_only)
```

Comment:
Why bother with these lambdas?  Why not just pass the variable?

Reply:
My original iteration of this PR had a call that was like
`buildbot.FailUnless(lambda: IsWindows() and 'lld' in name)` i.e. a more complex condition. It made that use case much nicer at a pretty small complexity cost.